diff --git a/dist/runtime/server/render/util.js b/dist/runtime/server/render/util.js
--- a/dist/runtime/server/render/util.js
+++ b/dist/runtime/server/render/util.js
@@ -156,6 +156,6 @@ function createBufferedRenderer(destination, renderFunction) {
 function createBufferedRenderer(destination, renderFunction) {
   return new BufferedRenderer(destination, renderFunction);
 }
-const isNode = typeof process !== "undefined" && Object.prototype.toString.call(process) === "[object process]";
+const isNode = typeof process !== "undefined" && Object.prototype.toString.call(process) === "[object process]" && !(typeof navigator !== "undefined" && navigator.userAgent === "Cloudflare-Workers");
 const isDeno = typeof Deno !== "undefined";
 function promiseWithResolvers() {
